src/Telemetry/Samples/FSharp/Program.fs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+open System
+open Microsoft.Quantum.Telemetry
+
+type SampleUnionType =
+    | SampleEnumValue1
+    | SampleEnumValue2
+
+type ExecutionCompleted =
+    {
+        SampleDateTime: DateTime
+        SampleString: string
+        SampleBool: bool
+        SampleEnum: SampleUnionType
+        [<PiiData>]
+        SamplePII: string
+        [<SerializeJson>]
+        SampleArray: string []
+        SampleTimeSpan: TimeSpan
+        SampleInt: int
+        SampleGuid: Guid
+        [<SerializeJson>]
+        SampleObjectToBeSerialized: Map<string, string>
+        SampleObjectToBeIgnored: obj
+        SampleException: Exception
+        SampleOptionWithValue: int option
+        SampleOptionWithNone: int option
+    }
+
+// The following 3 functions are just an artifact to
+// simulate the raise of a exception with a call stack
+let throwAnException throw : Unit =
+    if throw then raise (System.IO.FileNotFoundException(@"File path 'C:\Users\johndoe\file.txt'"))
+
+let throwANestedException throw = throwAnException throw
+
+let createExceptionWithStackTrace throw : Exception =
+    try
+        throwANestedException throw
+        null
+    with
+    | ex -> ex
+
+[<EntryPoint>]
+let main args =
+    // Initialize the TelemetryManager
+    let telemetryConfig =
```

Comment:
telemetry should only be enabled and initialized if the `TELEMETRY` compiler flag is set.

Reply:
Do we really need this TELEMETRY compiler constant?
Why not always build with telemetry included and opt-out during runtime if needed?

Reply:
Accepted you suggestion as discussed.